packages/@uppy/dashboard/src/GlobalSearch.browser.test.ts

```diff
@@ -0,0 +1,469 @@
+import Uppy from '@uppy/core'
+import Dashboard from '@uppy/dashboard'
+import RemoteSources, { type AvailablePluginsKeys } from '@uppy/remote-sources'
+import { page, userEvent } from '@vitest/browser/context'
+import {
+  afterAll,
+  afterEach,
+  beforeAll,
+  beforeEach,
+  describe,
+  expect,
+  test,
+} from 'vitest'
+import { worker } from './setup.js'
+
+import '@uppy/core/css/style.css'
+import '@uppy/dashboard/css/style.css'
+
+let uppy: Uppy
+
+beforeAll(async () => {
+  await worker.start({
+    onUnhandledRequest: 'bypass',
+    quiet: false,
+  })
+})
+
+afterAll(async () => {
+  await worker.stop()
+})
+
+function initializeUppy(sources: AvailablePluginsKeys[] = ['Dropbox']) {
+  document.body.innerHTML = '<div id="app"></div>'
+
+  return new Uppy({ id: 'uppy-e2e' })
+    .use(Dashboard, {
+      target: '#app',
+      inline: true,
+      height: 500,
+    })
+    .use(RemoteSources, {
+      companionUrl: 'http://companion.test',
+      sources,
+    })
+}
+
+beforeEach(() => {
+  uppy = initializeUppy(['Dropbox'])
+})
+
+afterEach(async () => {
+  if (uppy) {
+    // Small delay to allow Preact cleanup hooks to complete
+    await new Promise((resolve) => setTimeout(resolve, 50))
+    uppy.destroy()
+  }
+})
+
+describe('ProviderView Search E2E', () => {
+  test('Search for nested file in Dropbox and verify results', async () => {
+    await expect.element(page.getByText('My Device')).toBeVisible()
+    await page.getByRole('tab', { name: /dropbox/i }).click()
+    await expect.element(page.getByText('Import from Dropbox')).toBeVisible()
+    await expect.element(page.getByText('test-user@example.com')).toBeVisible()
+    await expect.element(page.getByText('first')).toBeVisible()
+
+    const searchInput = document.querySelector(
+      '.uppy-ProviderBrowser-searchFilterInput',
+    ) as HTMLInputElement
+    expect(searchInput).toBeDefined()
+
+    await userEvent.type(searchInput, 'target')
+
+    // Wait for debounce (500ms) + network response
+    await new Promise((resolve) => setTimeout(resolve, 600))
```

Comment:
again, we should probably disable debouncing for tests. 600ms is a long time to wait for each tests, as it adds up. AFAIK vitest browser mode and playwright already waits for UI to settle when looking for things.

Reply:
added.